activerecord/lib/active_record/relation/batches.rb

```diff
@@ -218,7 +218,7 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
           yielded_relation = where(primary_key => ids)
           yielded_relation.load_records(records)
         else
-          ids = batch_relation.pluck(primary_key)
+          ids = uncached { batch_relation.pluck(primary_key) }
```

Comment:
Let's chase `uncached_value` through `relation/{calcuations,finder_methods}.rb`.

We won't make it public API just now, but this still seems the best time to give it full coverage of queries the relation can make.